lib/nolibc/string.c

```diff
@@ -152,17 +152,21 @@ size_t strnlen(const char *str, size_t len)
 
 char *strncpy(char *dst, const char *src, size_t len)
 {
-	size_t clen;
-
-	clen = strnlen(src, len);
-	memcpy(dst, src, clen);
+	if (len != 0) {
+		char *d = dst;
+		const char *s = src;
+
+		do {
```

Comment:
I have a suggestion here: could we move the zeroing part of the destination after copying the effective data?

Something like:

```c
 while (--len != 0 && (*d++ = *s++) != 0));
 /* NUL pad the remaining n-1 bytes */
 while (--len != 0)
	*d++ = 0;
```

Would it affect the performance in some way?

Reply:
I don't think it would make any difference, it's worth noting that this is a somewhat inneficcient strncpy, as it copies things in bytes instead of words, an efficient strncpy is a bit of a monstruosity and don't think it belongs in nolibc

Reply:
I would go for the shortest approach, since it's more easier to follow, in my opinion. @razvand  do you have any opinion regarding this?

Reply:
Heya, sorry for the delay I was on holidays.
I'd keep as it is, since it's tested and I don't see an advantage in changing it.